sdk/search/azure-search-documents/azure/search/documents/indexes/_internal/_utils.py

```diff
@@ -317,8 +317,12 @@ def pack_search_indexer_data_source(search_indexer_data_source):
     # type: (SearchIndexerDataSourceConnection) -> _SearchIndexerDataSource
     if not search_indexer_data_source:
         return None
+    if search_indexer_data_source.connection_string is None or search_indexer_data_source.connection_string == "":
```

Comment:
When does this pack method run, though? We don't want to set it to "\<unchanged\>" in the response model, as that isn't actually correct. It's an implementation detail that probably shouldn't be exposed as the connection string. Having a response model with this value would be misleading.

Reply:
We expose a SearchIndexerDataSourceConnection model which has flattened DataSourceCredentials. 

Before we send the create_or_update request to the server, we call this method to translate it into a server-understandable SearchIndexerDataSourceConnection.

So in short, this method only update request. It does not touch response.